### ISSUE TITLE
[0.5.1] Cannot change logo in instance config when instance was provisioned with custom logo at install-time

### ISSUE THREAD

**Issue by emkll** [freedomofpress/securedrop#2853]
# Bug

## Description
SecureDrop logo is the file at `/var/www/securedrop/static/i/logo.png`

When using the default SecureDrop logo at install-time, permissions are
```
# ls -la /var/www/securedrop/static/i/logo.png
-rw-r--r-- 1 www-data www-data   864 Nov 21 23:35 logo.png
```
When the `securedrop_header_image` variable in `group_vars/all/site-specific` is set with a different image, files are uploaded with different permissions
```
# ls -la /var/www/securedrop/static/i/logo.png
-r-------- 1 www-data www-data 12506 Jan 13 13:49 logo.png
```
As such, the Instance Configuration tool cannot update the image as permissions on `/var/www/securedrop/static/i/logo.png` and an `IOError` is thrown`:
```
[Sat Jan 13 13:19:41.407386 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750] [2018-01-13 13:19:41,404] ERROR in app: Exception on /admin/config [POST]
[Sat Jan 13 13:19:41.407632 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750] Traceback (most recent call last):
[Sat Jan 13 13:19:41.407715 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]   File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1982, in wsgi_app
[Sat Jan 13 13:19:41.407790 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]     response = self.full_dispatch_request()
[Sat Jan 13 13:19:41.407865 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]   File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1614, in full_dispatch_request
[Sat Jan 13 13:19:41.407939 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]     rv = self.handle_user_exception(e)
[Sat Jan 13 13:19:41.408009 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]   File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1517, in handle_user_exception
[Sat Jan 13 13:19:41.408082 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]     reraise(exc_type, exc_value, tb)
[Sat Jan 13 13:19:41.408153 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]   File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1612, in full_dispatch_request
[Sat Jan 13 13:19:41.408225 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]     rv = self.dispatch_request()
[Sat Jan 13 13:19:41.408295 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]   File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1598, in dispatch_request
[Sat Jan 13 13:19:41.408370 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]     return self.view_functions[rule.endpoint](**req.view_args)
[Sat Jan 13 13:19:41.408442 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]   File "/var/www/securedrop/journalist_app/decorators.py", line 14, in wrapper
[Sat Jan 13 13:19:41.408513 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]     return func(*args, **kwargs)
[Sat Jan 13 13:19:41.408585 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]   File "/var/www/securedrop/journalist_app/admin.py", line 36, in manage_config
[Sat Jan 13 13:19:41.408657 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]     f.save(static_filepath)
[Sat Jan 13 13:19:41.408727 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]   File "/usr/local/lib/python2.7/dist-packages/werkzeug/datastructures.py", line 2703, in save
[Sat Jan 13 13:19:41.408800 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750]     dst = open(dst, 'wb')
[Sat Jan 13 13:19:41.408870 2018] [:error] [pid 1222:tid 2687130289920] [remote 127.0.0.1:35750] IOError: [Errno 13] Permission denied: '/var/www/securedrop/static/i/logo.png'
```
## Steps to Reproduce

- Set custom logo in `group_vars/all/site-specific` via `securedrop_header_image`
- Provision SecureDrop 0.5 prod vms
- observe permissions of `/var/www/securedrop/static/i/logo.png` on app server
- Update to 0.5.1-rc3
- observe permissions of `/var/www/securedrop/static/i/logo.png`
- Navigate to Journalist interface, Admin, instance config and attempt to change logo
- An error 500 occurs.

## Expected Behavior

Image should be changed and error 500 should not occur
## Actual Behavior

Image is not changed and error 500 occurs.
## Comments
If this bug is reproducible, I think it will need to be fixed in 2 distinct places:

- provisioning logic should be fixed to have consistent permissions at install time
- 0.5.1 postinst script should `chmod 644 /var/www/securedrop/static/i/logo.png`

**Comment by dachary** [comment]
This bug is actually consistently detected by travis (see [this run](https://travis-ci.org/freedomofpress/securedrop/builds/328410730?utm_source=github_status&utm_medium=notification)). Meaning we must **not** consider travis optional.

**Comment by dachary** [comment]
Good catch :-)

**Comment by redshiftzero** [comment]
Closed by #2854
